src/aap_eda/api/views/eda_credential.py

```diff
@@ -277,3 +281,43 @@ def destroy(self, request, *args, **kwargs):
             )
         self.perform_destroy(eda_credential)
         return Response(status=status.HTTP_204_NO_CONTENT)
+
+    @extend_schema(
+        description="Copy an EDA credential.",
+        request=None,
+        responses={
+            status.HTTP_201_CREATED: OpenApiResponse(
+                serializers.EdaCredentialSerializer,
+                description="Return the copied EDA credential.",
+            ),
+            status.HTTP_404_NOT_FOUND: OpenApiResponse(
+                None, description="EDA credential not found."
+            ),
+        },
+    )
+    @action(methods=["post"], detail=True, rbac_action=Action.READ)
+    def copy(self, request, pk):
+        eda_credential = self.get_object()
+        post_data = build_copy_post_data(eda_credential)
+
+        serializer = serializers.EdaCredentialCreateSerializer(data=post_data)
```

Comment:
after the post_data is prepared, the rest is the same as create. `copy` and `create` should share the same logic.